In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [9]:
# Import the CSV file NSMES1988updated.csv file and create a new dataframe for working with Pandas.
data = pd.read_csv('NSMES1988updated3.csv')
data.head()

,visits,nvisits,ovisits,novisits,emergency,hospital,health,chronic,adl,region,age,gender,married,school,income,employed,insurance,medicaid
0,5,0,0,0,0,1,average,2,normal,other,6.9,male,yes,6,2.8810,yes,yes,no
1,1,0,2,0,2,0,average,2,normal,other,7.4,female,yes,10,2.7478,no,yes,no
2,13,0,0,0,3,3,poor,4,limited,other,6.6,female,no,10,0.6532,no,no,yes
3,16,0,5,0,1,1,poor,2,limited,other,7.6,male,yes,3,0.6588,no,yes,no
4,3,0,0,0,0,0,average,2,limited,other,7.9,female,yes,6,0.6588,no,yes,no


In [10]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 4406 entries, 0 to 4405
Data columns (total 18 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   visits     4406 non-null   int64  
 1   nvisits    4406 non-null   int64  
 2   ovisits    4406 non-null   int64  
 3   novisits   4406 non-null   int64  
 4   emergency  4406 non-null   int64  
 5   hospital   4406 non-null   int64  
 6   health     4406 non-null   str    
 7   chronic    4406 non-null   int64  
 8   adl        4406 non-null   str    
 9   region     4406 non-null   str    
 10  age        4406 non-null   float64
 11  gender     4406 non-null   str    
 12  married    4406 non-null   str    
 13  school     4406 non-null   int64  
 14  income     4406 non-null   float64
 15  employed   4406 non-null   str    
 16  insurance  4406 non-null   str    
 17  medicaid   4406 non-null   str    
dtypes: float64(2), int64(8), str(8)
memory usage: 619.7 KB


In [11]:
numurical_cols = data.select_dtypes(include=['int64','float64']).columns.tolist()
categorical_cols = data.select_dtypes(include=['object', 'category', 'bool']).columns.tolist()
print(numurical_cols)
print(categorical_cols)

['visits', 'nvisits', 'ovisits', 'novisits', 'emergency', 'hospital', 'chronic', 'age', 'school', 'income']
['health', 'adl', 'region', 'gender', 'married', 'employed', 'insurance', 'medicaid']


/var/folders/f7/6ttbslv14mb1smgbhyrcq1xh0000gn/T/ipykernel_28467/4133293364.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_cols = data.select_dtypes(include=['object', 'category', 'bool']).columns.tolist()


In [12]:
for col in  categorical_cols:
    print(f"{col}: {data[col].unique()}")

health: <StringArray>
['average', 'poor', 'excellent']
Length: 3, dtype: str
adl: <StringArray>
['normal', 'limited']
Length: 2, dtype: str
region: <StringArray>
['other', 'midwest', 'northeast', 'west']
Length: 4, dtype: str
gender: <StringArray>
['male', 'female']
Length: 2, dtype: str
married: <StringArray>
['yes', 'no']
Length: 2, dtype: str
employed: <StringArray>
['yes', 'no']
Length: 2, dtype: str
insurance: <StringArray>
['yes', 'no']
Length: 2, dtype: str
medicaid: <StringArray>
['no', 'yes']
Length: 2, dtype: str


In [13]:
# Perform analysis based on the following criteria: Different types of visits, Gender, Marital 
#  Status, School, Income, Employment Status, Insurance, Medical Aid

In [14]:
visit_cols = ['visits', 'nvisits', 'ovisits', 'novisits', 'emergency', 'hospital']

In [15]:
# Different types of visits — overall summary
for col in visit_cols :
    print("Visit types summary:\n", data[col].describe(), "\n")

Visit types summary:
 count    4406.000000
mean        5.774399
std         6.759225
min         0.000000
25%         1.000000
50%         4.000000
75%         8.000000
max        89.000000
Name: visits, dtype: float64 

Visit types summary:
 count    4406.000000
mean        1.618021
std         5.317056
min         0.000000
25%         0.000000
50%         0.000000
75%         1.000000
max       104.000000
Name: nvisits, dtype: float64 

Visit types summary:
 count    4406.000000
mean        0.750794
std         3.652759
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max       141.000000
Name: ovisits, dtype: float64 

Visit types summary:
 count    4406.000000
mean        0.536087
std         3.879506
min         0.000000
25%         0.000000
50%         0.000000
75%         0.000000
max       155.000000
Name: novisits, dtype: float64 

Visit types summary:
 count    4406.000000
mean        0.263504
std         0.703659
min         0.000000
25%   

In [16]:
# Visits by Gender
data.groupby('gender')[visit_cols].mean()

,visits,nvisits,ovisits,novisits,emergency,hospital
gender,,,,,,
female,6.014079,1.732116,0.683029,0.513318,0.269406,0.285769
male,5.420135,1.449381,0.850956,0.569741,0.254781,0.311024


In [17]:
# Visits by Marital Status
data.groupby('married')[visit_cols].mean()

,visits,nvisits,ovisits,novisits,emergency,hospital
married,,,,,,
no,5.931500,1.549000,0.65000,0.507000,0.30150,0.311000
yes,5.643807,1.675395,0.83458,0.560266,0.23192,0.283458


In [18]:
# School by Gender & Employment
data.groupby('gender')['school'].mean()

gender
female    10.257610
male      10.338583
Name: school, dtype: float64

In [19]:
data.groupby('employed')['income'].mean()

employed
no     2.360056
yes    3.977940
Name: income, dtype: float64

In [20]:
# Insurance vs Visits & Income
data.groupby('insurance')[['visits','income']].mean()

,visits,income
insurance,,
no,4.912690,1.662984
yes,6.022508,2.775944


In [21]:
# Medical Aid (Medicaid) vs Visits & Income
data.groupby('medicaid')[['visits','income']].mean()

,visits,income
medicaid,,
no,5.68007,2.666747
yes,6.71393,1.136538


In [ ]:
# Explore and analyze the dataset to gain insights into how different factors relate to each 
# other within the dataset. By grouping the data according to specific demographic and 
# economic criteria, create a series of distribution tables by considering the below 
# instructions


In [24]:
data['age'].unique()


array([ 6.9,  7.4,  6.6,  7.6,  7.9,  7.5,  8.7,  7.3,  7.8,  8.4,  8.1,
        8.3,  8. ,  6.7,  7.1,  8.9,  7.2,  9.6,  9.3,  7. ,  7.7,  6.8,
        9. ,  8.2,  8.6,  8.5,  9.7,  9.4,  8.8,  9.1,  9.5,  9.2, 10.2,
       10.9,  9.9,  9.8])

In [25]:
# Step 1: bin age into groups (age column is in years/10, e.g. 6.9 = 69 yrs)
data['age_years'] = data['age'] * 10
bins = [60, 70, 80, 90, 100, 110]
labels = ['60-69', '70-79', '80-89', '90-99', '100+']
data['age_group'] = pd.cut(data['age_years'], bins=bins, labels=labels, right=False)

In [26]:
# Step 2: count individuals per age group, split by gender
age_gender_dist = pd.crosstab(data['age_group'], data['gender'])
print(age_gender_dist)

gender     female  male
age_group              
60-69         723   554
70-79        1369   924
80-89         475   265
90-99          59    34
100+            2     1


In [28]:
health_gender_dist = pd.crosstab(data['health'], data['gender'])
print(health_gender_dist)

gender     female  male
health                 
average      2093  1416
excellent     193   150
poor          342   212


In [29]:
data.head()

,visits,nvisits,ovisits,novisits,emergency,hospital,health,chronic,adl,region,age,gender,married,school,income,employed,insurance,medicaid,age_years,age_group
0,5,0,0,0,0,1,average,2,normal,other,6.9,male,yes,6,2.8810,yes,yes,no,69.0,60-69
1,1,0,2,0,2,0,average,2,normal,other,7.4,female,yes,10,2.7478,no,yes,no,74.0,70-79
2,13,0,0,0,3,3,poor,4,limited,other,6.6,female,no,10,0.6532,no,no,yes,66.0,60-69
3,16,0,5,0,1,1,poor,2,limited,other,7.6,male,yes,3,0.6588,no,yes,no,76.0,70-79
4,3,0,0,0,0,0,average,2,limited,other,7.9,female,yes,6,0.6588,no,yes,no,79.0,70-79


In [31]:
# Income Distribution by Gender: Compile a table to examine the income distribution across genders
income_gend = data.groupby('gender')['income'].describe()
income_gend

,count,mean,std,min,25%,50%,75%,max
gender,,,,,,,,
female,2628.0,2.249348,2.721831,-1.0125,0.739275,1.4160,2.780550,54.8351
male,1778.0,2.937716,3.157305,-1.0125,1.253700,2.0574,3.565175,54.8351


In [34]:
# Regional Income Distribution: Prepare a table to display the income distribution across various regions
income_region = data.groupby('region')['income'].describe()
income_region

,count,mean,std,min,25%,50%,75%,max
region,,,,,,,,
midwest,1157.0,2.513634,3.107841,0.0000,1.01640,1.7875,3.159600,54.8351
northeast,837.0,2.679709,3.196229,0.0000,1.00500,1.7413,3.366400,41.7596
other,1614.0,2.166284,2.323474,-1.0125,0.78330,1.4220,2.698475,36.0024
west,798.0,3.116505,3.314857,-0.8180,1.01025,2.0656,4.070325,24.2161


In [ ]:
# Age-wise Income Analysis: Develop a table to analyze the relationship between age and income.

In [35]:
# reuse the age_group column created earlier with pd.cut()
age_income = data.groupby('age_group')['income'].mean()
age_income

age_group
60-69    2.784914
70-79    2.508397
80-89    2.162176
90-99    2.385023
100+     1.545567
Name: income, dtype: float64

# Capstone Session 3 — Data Analysis Report (NSMES1988 Healthcare Dataset)

## Dataset Overview
- **4406 records**, 18 columns, no missing values.
- **Numeric columns**: visits, nvisits, ovisits, novisits, emergency, hospital, chronic, age, school, income
- **Categorical columns**: health, adl, region, gender, married, employed, insurance, medicaid

## Categorical Variable Levels
| Column | Categories |
|---|---|
| health | average, poor, excellent |
| adl | normal, limited |
| region | other, midwest, northeast, west |
| gender | male, female |
| married | yes, no |
| employed | yes, no |
| insurance | yes, no |
| medicaid | no, yes |

## Visit Types — Overall Summary
Physician office visits (`visits`) dominate, averaging **5.77** per person (max 89). All other visit types (nvisits, ovisits, novisits, emergency, hospital) have medians of **0**, meaning most people never use those services — a small subset of high-utilizers drives the mean up (all show high std relative to mean → right-skewed/outlier-heavy).

## Visits by Gender
Females average **more visits** than males across most types:
- visits: female 6.01 vs male 5.42
- nvisits: female 1.73 vs male 1.45
- ovisits: male 0.85 vs female 0.68 (males slightly higher here)

## Visits by Marital Status
Unmarried individuals average slightly more visits (5.93) than married individuals (5.64), though nvisits/ovisits/hospital rates are close between groups.

## School (Education) by Gender
Nearly identical: female 10.26 years vs male 10.34 years — negligible gender gap in education.

## Income by Employment Status
Clear gap: employed individuals average **$3.98** (×10,000) vs **$2.36** for unemployed — confirms employment strongly drives income.

## Visits & Income by Insurance
Insured individuals have **both** higher visits (6.02 vs 4.91) and higher income (2.78 vs 1.66) than uninsured — insurance correlates with greater healthcare access.

## Visits & Income by Medicaid
Medicaid recipients have **higher visits** (6.71 vs 5.68) but **lower income** (1.14 vs 2.67) — consistent with Medicaid serving lower-income, higher-need populations.

## Age and Gender Distribution
| Age Group | Female | Male |
|---|---|---|
| 60-69 | 723 | 554 |
| 70-79 | 1369 | 924 |
| 80-89 | 475 | 265 |
| 90-99 | 59 | 34 |
| 100+ | 2 | 1 |

Females outnumber males in every age group, and the gap widens with age — consistent with longer average female life expectancy. Most individuals (both genders) fall in the 70-79 group.

## Health Status by Gender
| Health | Female | Male |
|---|---|---|
| average | 2093 | 1416 |
| excellent | 193 | 150 |
| poor | 342 | 212 |

Proportionally similar between genders (~80% average, ~7-8% excellent, ~12-13% poor) — no major gender disparity in self-reported health.

## Income Distribution by Gender
| Gender | Mean | Std | Min | Max |
|---|---|---|---|---|
| female | 2.25 | 2.72 | -1.01 | 54.84 |
| male | 2.94 | 3.16 | -1.01 | 54.84 |

Males average higher income than females (2.94 vs 2.25). Both genders share the same negative minimum (-1.01), a data-quality anomaly worth flagging.

## Regional Income Distribution
| Region | Mean | Max |
|---|---|---|
| west | 3.12 | 24.22 |
| northeast | 2.68 | 41.76 |
| midwest | 2.51 | 54.84 |
| other | 2.17 | 36.00 |

**West** has the highest average income; **"other"** the lowest — regional economic disparity is evident.

## Age-wise Income Analysis
| Age Group | Mean Income |
|---|---|
| 60-69 | 2.78 |
| 70-79 | 2.51 |
| 80-89 | 2.16 |
| 90-99 | 2.39 |
| 100+ | 1.55 |

Income **declines steadily with age** after 60-69, consistent with post-retirement income reduction, with a further drop at 100+.

## Key Observations
1. **Negative income values (-1.01, -0.82)** exist in the data — a cleaning issue to address before further modeling.
2. **Insurance and Medicaid status are the strongest differentiators** of both healthcare usage and income among all factors examined.
3. **Gender differences are modest** overall (visits, education, health) except in income, where males average notably higher.
4. **Age is inversely related to income**, and the population skews older (majority in 70-79 group) and female-heavy in older brackets.
5. **Region matters economically** — West residents earn most on average; "other" region residents earn least.
